README.md

```diff
@@ -1,6 +1,32 @@
-ooc-kean
+magic-sdk
 ========
+**Last updated: 2016-01-18**
 
-ooc port of parts of Kean
+This is an extended SDK for `magic`, a high-level cross-platform language which transpiles into C99.
 
-Uses [Sean Barrett's stb library](https://github.com/nothings/stb) for opening and saving images.
+## History
+This project was originally a fork of the [ooc project](https://github.com/fasterthanlime/rock) and contains ports of selected parts of the C# library [Kean](https://github.com/cogneco/Kean). The most notable changes include dropping garbage collection. It also uses [Sean Barrett's stb library](https://github.com/nothings/stb) for opening and saving images.
+
+## How to run
+To compile, you need the Cogneco version of the [Rock compiler](https://github.com/cogneco/rock). To make life easy, you might also want an editor with ooc/magic syntax highlighting support and a style enforcer. Those are available (for Ubuntu) at the [magic-tools repo](https://github.com/cogneco/magic-tools#installation-ubuntu), which can also be used to automatically install (and update) rock.
+
+Tests can be run using the `test.sh` script. Compiling anything using this SDK requires the `--gc=off` flag to be passed along to `rock`.
+
+## How to learn magic
+There are several original sources:
+
+- [The original ooc documentation](https://ooc-lang.org/docs/)
+- [The original ooc tutorial](https://ooc-lang.org/docs/tutorial/)
+- [Wikibooks](https://en.wikibooks.org/wiki/Programming_with_ooc)
+
+The above however target the original ooc language and are as such getting more and more outdated. A slightly more updated version is available [here](http://oocmanual.cogneco.com/language/tuples/). **TODO**
+
```

Comment:
Manual may not be hosted on cogneco.com in the future so maybe we should have a link directly to the Github repository.